pages/[username]/[collectionId]/[tokenId]/index.tsx

```diff
@@ -22,7 +23,7 @@ export default function NftDetailPage({ username, collectionId, tokenId }: NftDe
 
   // the default "back" behavior from the NFT Detail Page
   // is a redirect to the Collection Page
-  const collectionRoute = `/${username}/${collectionId}`;
+  const collectionRoute = ROUTES.USER.COLLECTION(username, collectionId);
```

Comment:
curious where this pattern is from? while i can see how this offers safer guarantees, i think the original format makes the route super clear! as opposed to having a level of indirection

Reply:
Good feedback! For me personally this actually reads fine and provides a somewhat scalable way to make the routes. But, yeah this could get pretty long and perhaps not as clear.

Another option is to use a function that would sit in the routes constant file:
```
export function userCollectionRoute(username: string, collectionId: string) {
  return `${ROUTES.USER.ROOT(username)}/${collectionId}`;
}
```

It's shorter, and still has the same guarantees.